### ISSUE TITLE
[Feature] Support multiple positional input files in CLI (`ys [OPTIONS] [FILES...]`)

### ISSUE THREAD

**Issue by aolieman** [yaml-schema/yaml-schema#75]
### Summary

Currently, `ys` accepts only a single target file positional argument:
`ys [OPTIONS] [FILE] [COMMAND]`

When attempting to validate multiple files in one go (e.g. via shell globs or `pre-commit` batching), `ys` fails with an unexpected argument error:

```txt
$ ys -f schema.yml file1.yml file2.yml
error: unexpected argument 'file2.yml' found
```

### Proposed Solution

Update `clap` / CLI argument parsing to accept multiple positional file arguments (e.g. `Vec<PathBuf>` or `[FILES...]`).

### Benefits

1. **Performance:** The schema file can be read and compiled once in memory, then validated against all input files in a loop (or in parallel using `rayon`).
2. **Standard CLI Compatibility:** Matches standard UNIX tool conventions and popular linters.
3. **Native Pre-Commit Support:** Allows `ys` to function seamlessly as a `pre-commit` hook without needing shell wrapper scripts, see #71.

**Comment by aisrael** [comment]
Hi, @aolieman! This is a great suggestion, thanks.

One question though... Currently `ys` supports specifying a single schema (`-f schema.yml`), or, if not, loading and validating against a `$schema` directive (if present) in the YAML file itself. If neither are provided, it exits with an error.

With multiple files, we'll extend this such that if `-f schema.yml` is provided, that schema will be used to validate all the input files. Otherwise, we'll try to find and load the `$schema` for each individual file.

Does that sound good?

**Comment by aolieman** [comment]
Hi @aisrael, thank you for working on this enhancement!

> With multiple files, we'll extend this such that if `-f schema.yml` is provided, that schema will be used to validate all the input files. Otherwise, we'll try to find and load the `$schema` for each individual file.

Yes, that sounds excellent.

I've quickly scanned the plan md in #76 and that looks good to me as well.
